reconcile/jenkins_job_builder.py

```diff
@@ -182,10 +194,18 @@ def validate_repos_and_admins(jjb, additional_repo_urls):
 
 
 @defer
-def run(dry_run, io_dir='throughput/', defer=None):
-    jjb, additional_repo_urls = init_jjb()
+def run(dry_run, io_dir='throughput/', print_only=False,
+        config_name=None, job_name=None, instance_name=None, defer=None):
+    if not print_only and config_name:
+        raise Exception("--config-name must works with --print-only mode")
```

Comment:
This PR introduces a third working mode for jenkins-job-builder and three flitter level.

flitter | print-only | dry-run | no-dry-run
------------ | -------------  | -------------  | -------------
config_name | print jobs defined in exact config file | generate desired XML file for jobs defined in exact config file and try to delete most other jobs [1] | delete most other jobs [2]
job_name | print jobs match with job_name | no effect  | no effect
instance_name | print jobs for exact instance | compare jobs for exact instance | update jobs for exact instance

This line is to avoid running into case [1] and [2]